apm-agent-core/src/test/java/co/elastic/apm/agent/report/ApmServerReporterIntegrationTest.java

```diff
@@ -136,17 +164,53 @@ void testSecretToken() throws ExecutionException, InterruptedException {
             exchange.setStatusCode(200).endExchange();
         };
         reporter.report(new Transaction(tracer));
-        reporter.flush().get();
+        reporter.flush();
         assertThat(reporter.getDropped()).isEqualTo(0);
         assertThat(receivedIntakeApiCalls.get()).isEqualTo(1);
     }
 
     @Test
-    void testReportErrorCapture() throws ExecutionException, InterruptedException {
+    void testReportErrorCapture() {
         reporter.report(new ErrorCapture(tracer));
-        reporter.flush().get();
+        reporter.flush();
+        assertThat(reporter.getDropped()).isEqualTo(0);
+        assertThat(receivedIntakeApiCalls.get()).isEqualTo(1);
+    }
+
+    @Test
+    void testTimeout() {
+        doReturn(TimeDuration.of("1ms")).when(reporterConfiguration).getApiRequestTime();
+        doAnswer(invocation -> System.nanoTime()).when(clock).nanoTicks();
+        reporter.report(new Transaction(tracer));
+        await().untilAsserted(() -> assertThat(reporter.getReported()).isEqualTo(1));
         assertThat(reporter.getDropped()).isEqualTo(0);
         assertThat(receivedIntakeApiCalls.get()).isEqualTo(1);
+        assertThat(receivedEvents.get()).isEqualTo(2);
+    }
+
+    @Test
+    void testFlush() {
+        reporter.report(new Transaction(tracer));
+        assertThat(receivedEvents.get()).isEqualTo(0);
+        assertThat(reporter.flush(5, TimeUnit.SECONDS)).isTrue();
+        assertThat(reporter.getDropped()).isEqualTo(0);
+        assertThat(reporter.getReported()).isEqualTo(1);
+    }
+
+    @Test
+    void testFailingApmServer() {
+        statusCode = HttpStatus.SERVICE_UNAVAILABLE_503;
+        // try to report a few events to trigger backoff
+        reporter.report(new Transaction(tracer));
+        reporter.flush(1, TimeUnit.SECONDS);
+        reporter.report(new Transaction(tracer));
+        reporter.flush(1, TimeUnit.SECONDS);
+        reporter.report(new Transaction(tracer));
+        reporter.flush(1, TimeUnit.SECONDS);
+        reporter.report(new Transaction(tracer));
+
+        assertThat(v2handler.isHealthy()).isFalse();
+        assertThat(reporter.flush(1, TimeUnit.SECONDS)).isFalse();
```

Comment:
```suggestion
        assertThat(reporter.flush(0, TimeUnit.SECONDS)).isFalse();
```
Should be shortcut by the fact that `isHealthy() == false`, right?